src/state/slices/opportunitiesSlice/resolvers/foxFarming/index.ts

```diff
@@ -126,7 +130,9 @@ export const foxFarmingLpMetadataResolver = async ({
         ] as const,
         name: LP_EARN_OPPORTUNITIES[opportunityId].opportunityName,
       },
-    } as OpportunitiesState[DefiType.LiquidityPool]['byId'],
+    } as Partial<
```

Comment:
Taken from `OpportunitiesState[DefiType.LiquidityPool]['byId']`, we might want not to partial it in a follow-up PR, given this effectively loses the opportunitySpecific discrimination: tsc is happy if it's there, but doesn't complain if it isn't. TSC playground confirmed the types are correct, which mean `Partial<>` effectively voids our guarantees here

Reply:
I don't think we want to typecast here.

This block should probably be:

```ts
const data: GetOpportunityMetadataOutput<DefiProvider.FoxFarming, DefiType.LiquidityPool> = {
    byId: {
      [opportunityId]: {
        apy,
        assetId: opportunityId,
        provider: DefiProvider.FoxFarming as const,
        tvl,
        type: DefiType.LiquidityPool as const,
        underlyingAssetId: foxEthLpAssetId,
        underlyingAssetIds: foxEthPair,
        underlyingAssetRatios: [
          toBaseUnit(ethPoolRatio.toString(), assets.byId[foxEthPair[0]]?.precision ?? 0),
          toBaseUnit(foxPoolRatio.toString(), assets.byId[foxEthPair[1]]?.precision ?? 0),
        ] as const,
        name: LP_EARN_OPPORTUNITIES[opportunityId].opportunityName,
      },
    },
    type: opportunityType,
  }
```

Reply:
Much better, ty! No assertion, no partial and properly leveraging the newly added generics 🤌 
3a39e20ea138319032420d79d970c7900e3c167f